index.js

```diff
@@ -63,17 +65,49 @@ module.exports.makeApp = function () {
   app.use(cors())
   app.use(cookieParser())
   app.use(i18n.init)
-  app.use(session({
+
+  // Default session options
+  let sessionOptions = {
     secret: config.get('SESSION_SECRET'),
-    resave: true,
     rolling: true,
     saveUninitialized: false,
     cookie: {
       maxAge: config.get("SESSION_COOKIE_MAX_AGE")
         ? parseInt(config.get("SESSION_COOKIE_MAX_AGE"))
         : 60 * 60 * 1000
     }
-  }))
+  }
+
+  // Redis session store for production
+  if (config.get('REDIS_URL')) {
+    const redisClient = redis.createClient({
+      url: config.get('REDIS_URL')
+    })
+    
+    redisClient.on('error', err => {
+      logger.error(err);
+    });
+
+    sessionOptions.resave = false
+    sessionOptions.store = new RedisStore({ client: redisClient })
+  }
+
+  app.use(session({ ...sessionOptions }))
+
+  // Handling session error
+  app.use(function (req, res, next) {
+    if (!req.session) {
+      var err = new Error();
+      err.status = 500;
+      err.statusText = 'SESSION EXPIRED OR NOT FOUND'
+      err.text = () => {
+        return 'Failed to configure session.'
+      }
+      next(err)
```

Comment:
@sagargg does that mean the app will through 500 if there is no session?

Reply:
@sagargg can we update status text to something more user-friendly, eg:

> Something has failed. Please, try again later.

Reply:
@anuveyatsu  updated.